### PR TITLE
Sensitive content is now hidden from the recent apps when app lock is enabled

### PR DESCRIPTION
**Pull request by iskandergaba**
- Added the requested feature #602. Sensitive content (note list, tag list, note content, and settings) are now hidden from recent apps when app lock is enabled.
- Some code clean up along the way.

**App lock enabled:** Non-snesitive content screen

screenshot_1

**App lock enabled:** Snesitive content screen

screenshot_2

@roundhill Please check this out and let me know.

### CODE REVIEW

**Comment by iskandergaba** [comment]
@roundhill So basically I am specifying the flag `FLAG_Secure` **in the activities containing sensitive information when the application lock is enabled**. This is a common practice that banking applications and privacy centered applications like Firefox Focus follow to hide the thumbnail of the application from `Recent Apps` menu. The problem with this approach is that it disables screenshots from being taken on that activity. So basically, when the application lock is enabled, thumbnail from `Recent Apps` are hidden and no screenshot can be taken on those sensitive information activities. When the application lock is disabled, the behavior is normal.

I tried to add the flag for instance in `OnPause()` (To hide thumbnail in `Recent Apps`) and remove it in `OnResume()` (to enable screenshot functionality) method for instance to get the exact desired behavior (remove thumbnail but keep screenshot functionality) without sacrificing taking screenshots, which worked a bit but inconsistently I am afraid. The lifecycle calls are not guaranteed to be made synchronously and therefore rarely this worked. I played with a lot of other combinations lifecycle methods but in vain. So my question here is: Is it fine to sacrifice the functionality of taking screenshots on sensitive information activities when the application is locked or not? If so, I will do a little tweak to this approach and will be good to go. Otherwise, I am afraid there is not any other way of doing this unless we cover the layout with another layout before the activity is paused (which is not so cool I think).

Let me know so that I can proceed.

**Comment by roundhill** [comment]
> Is it fine to sacrifice the functionality of taking screenshots on sensitive information activities when the application is locked or not?

Sounds OK to me. Would we be able to show a Toast or something if a screenshot is attempted that it is disabled because the passcode lock is enabled?

**Comment by iskandergaba** [comment]
@roundhill I am not sure how to listen to when the user attempts to take a screenshot in a clean and effective way so that I show a toast when such an event occurs.

An easier indication however would be to add description to `Turn PIN Lock on` setting that describes this behavior briefly.

What do you think?

**Comment by roundhill** [comment]
> An easier indication however would be to add description to `Turn PIN Lock on` setting that describes this behavior briefly.

Sure, let's go for that 👍

**Comment by iskandergaba** [comment]
> Sure, let's go for that 👍
Done.

**Comment by iskandergaba** [comment]
> Sure, let's go for that 👍

Done.

**Comment by iskandergaba** [comment]
> Sure, let's go for that 👍

Done.

screenshot

**Comment by roundhill** [comment]
Great. How about shortening it to `App screenshots will be disabled while this option is on.`?

**Comment by iskandergaba** [comment]
> Great. How about shortening it to `App screenshots will be disabled while this option is on.`?

Sure. Here you go.

screenshots

**Comment by roundhill** [comment]
Sorry for the delay in reviewing this @iskandergaba! 

I was a little confused at first because this wasn't working after I enabled the pin lock in `NotesActivity`. I think it's because it isn't calling `onCreate` again because the activity already exists. It works great after I restarted the app though 👍Any workarounds for that?

**Comment by roundhill** [comment]
> Let me know how it tests on yours.

It still isn't working for me, unfortunately.

1. Disable PIN lock.
2. Enable PIN lock, and return to notes activity.
3. Press multitask button, observe notes list is still visible.

**Comment by iskandergaba** [comment]
@roundhill I am not sure what is going wrong. I tested many times using my Galaxy S6 Edge and on emulated Pixel 2 Device and it worked like a charm :/

**Comment by roundhill** [comment]
> I tested many times using my Galaxy S6 Edge and on emulated Pixel 2 Device and it worked like a charm :/

OK, I'll check again!

File: Simplenote/src/main/java/com/automattic/simplenote/TagsActivity.java [hunk 1]
```diff
@@ -41,4 +43,16 @@ public void onCreate(Bundle savedInstanceState) {
...
+    @Override
+    protected void onResume() {
+        super.onResume();
+
+        // Disable screenshots if app is lock is on
+        if (AppLockManager.getInstance().getAppLock().isPasswordLocked()) {
+            getWindow().addFlags(WindowManager.LayoutParams.FLAG_SECURE);
+        } else {
+            getWindow().clearFlags(WindowManager.LayoutParams.FLAG_SECURE);
+        }
```
[the remark this answers is not in the document]
Comment (iskandergaba) [reply Simplenote/src/main/java/com/automattic/simplenote/TagsActivity.java]: Got it. Will do it tonight.

**Comment by iskandergaba** [comment]
@theck13 I implemented the changes you requested.

File: Simplenote/src/main/java/com/automattic/simplenote/utils/DisplayUtils.java [hunk 2]
```diff
@@ -72,8 +75,15 @@ public static int getChecklistIconSize(Context context) {
         if (context == null) {
             return 18;
         }
-
         return DisplayUtils.dpToPx(context, PrefUtils.getFontSize(context)) + DisplayUtils.dpToPx(context, 6);
     }
-}
 
+  // Disable screenshots if app is lock is on
```
Comment (theck13) [inline Simplenote/src/main/java/com/automattic/simplenote/utils/DisplayUtils.java]:
```suggestion
  // Disable screenshots if app PIN lock is on
```

I think this comment has a typographical error.  Should it be "// Disable screenshots if app PIN lock is on" instead?

Comment (iskandergaba) [reply Simplenote/src/main/java/com/automattic/simplenote/utils/DisplayUtils.java]: Yes you are right. Sorry it slipped my mind.

**Pull request merged**